pkg/webhook/util/crd/crd.go

```diff
@@ -49,41 +52,61 @@ func Ensure(client apiextensionsclientset.Interface, lister apiextensionslisters
 		return fmt.Errorf("failed to list crds: %v", err)
 	}
 
-	webhookConfig := apiextensionsv1.WebhookClientConfig{
-		CABundle: caBundle,
-	}
-	path := "/convert"
-	if host := webhookutil.GetHost(); len(host) > 0 {
-		url := fmt.Sprintf("https://%s:%d%s", host, webhookutil.GetPort(), path)
-		webhookConfig.URL = &url
-	} else {
-		var port int32 = 443
-		webhookConfig.Service = &apiextensionsv1.ServiceReference{
-			Namespace: webhookutil.GetNamespace(),
-			Name:      webhookutil.GetServiceName(),
-			Port:      &port,
-			Path:      &path,
-		}
-	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.EnableExternalCerts) {
+		for _, crd := range crdList {
+			if len(crd.Spec.Versions) == 0 || crd.Spec.Conversion == nil || crd.Spec.Conversion.Strategy != apiextensionsv1.WebhookConverter {
+				continue
+			}
+			if !kruiseScheme.Recognizes(schema.GroupVersionKind{Group: crd.Spec.Group, Version: crd.Spec.Versions[0].Name, Kind: crd.Spec.Names.Kind}) {
+				continue
+			}
+
+			if crd.Spec.Conversion.Webhook == nil || crd.Spec.Conversion.Webhook.ClientConfig == nil {
+				return fmt.Errorf("bad conversion configuration of CRD %s", crd.Name)
+			}
 
-	for _, crd := range crdList {
-		if len(crd.Spec.Versions) == 0 || crd.Spec.Conversion == nil || crd.Spec.Conversion.Strategy != apiextensionsv1.WebhookConverter {
-			continue
+			if !bytes.Equal(crd.Spec.Conversion.Webhook.ClientConfig.CABundle, caBundle) {
+				return fmt.Errorf("caBundle of CRD %s does not match external caBundle", crd.Name)
+			}
+		}
```

Comment:
just return in the end of if clause and remove the else to make the indentation smaller